promql/value.go

```diff
@@ -168,6 +168,21 @@ func (p HPoint) MarshalJSON() ([]byte, error) {
 	return json.Marshal([...]interface{}{float64(p.T) / 1000, h})
 }
 
+// histogramSamples returns the number of samples representing the histogram.
+// each sample is 4 bytes.
+func (p HPoint) histogramSamples() int {
+	return p.H.Size() / 4
```

Comment:
I think it should be `(p.H.Size() + 8) / 16`.

In `TotalSamples` below, we count each `FPoint` as "one sample". One `FPoint` is 16 bytes, a 64bit int timestamp and a 64bit float sample value. So "one sample" so far has been 16 bytes.

For an `HPoint`, we have the same timestamp (8 bytes), and then a number of bytes in the histogram, the sum of which we need to divide by the 16 bytes we considered "one sample" so far.